modules/client-handler/src/main/java/org/apache/ignite/client/handler/ClientContext.java

```diff
@@ -79,6 +94,20 @@ public BitSet features() {
         return features;
     }
 
+    /**
+     * Gets the authentication request.
+     *
+     * @return Authentication request.
+     */
+    public AuthenticationRequest<?, ?> authenticationRequest() {
```

Comment:
We must not store an authentication request in the client context. There is a password that the user passed into the system. The system must not store this value. 

I think we can just close the connection instead of re-authenticate it on the fly.